src/Services/Actions/AsyncActions.js

```diff
@@ -0,0 +1,26 @@
+import { fetchVideosAction } from '.';
+
+// eslint-disable-next-line import/prefer-default-export
+export const fetchVideosFromServer = () => (dispatch) => {
+  fetch('http://localhost:4000/movies?sortBy=release_date&sortOrder=desc&limit=6')
+    .then((response) => response.json())
+    .then((json) => dispatch(fetchVideosAction(json)));
+};
+
+export const fetchVideosSortBy = (field) => (dispatch) => {
```

Comment:
you have a bunch of functions that are generated for each of the change... the flow in app is event driven and because of that we have too much of variations for just one movies request

If we do approach state driven,  it will be more automated. Meaning, if we have one global state, where sortBy and genre selection is stored, we could add useEffect that will react on their change and trigger fetch movies action with this selection -> url is generated considering both genre and sortBy, and this will be just one single request variation

let's discuss it offline if it sounds confusing